### ISSUE TITLE
Panic when field name includes a `:`

### ISSUE THREAD

**Issue by boydgreenfield** [vega/vegafusion#485]
When a field name includes a `:`, `vegafusion` cannot handle it and raises a `PanicException`. Here's a minimal example for which the `spec` works fine in the Vega Editor:

```python3
import vegafusion as vf
import vl_convert as vlc

spec = {
  "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
  "data": {"values": [        
        {"Horsepower:": 225, "Miles_per_Gallon": 14, "Origin": "USA"},
        {"Horsepower:": 95, "Miles_per_Gallon": 24, "Origin": "Japan"},
        {"Horsepower:": 95, "Miles_per_Gallon": 22, "Origin": "USA"},
        {"Horsepower:": 97, "Miles_per_Gallon": 18, "Origin": "USA"},
      ]},
  "transform": [{
    "calculate": "datum.Origin[0]",
    "as": "OriginInitial"
  }],
  "mark": "text",
  "encoding": {
    "x": {"field": "Horsepower:", "type": "quantitative"},
    "y": {"field": "Miles_per_Gallon", "type": "quantitative"},
    "color": {"field": "Origin", "type": "nominal"},
    "text": {"field": "OriginInitial", "type": "nominal"}
  }
}

vlc.vegalite_to_vega(spec)  # works
vf.runtime.build_pre_transform_spec_plan(vlc.vegalite_to_vega(spec))  # raises PanicException
```

**Comment by jonmmease** [comment]
Can confirm that this panics in VegaFusion and works in Vega-Lite. Thanks for the report!

**Comment by Keats** [comment]
So this is not actually due to the colon at all, you can remove them and it will still panic.
The actual error is `ValueError: Failed to parse chart_spec dict as Vega: unexpected type: 'dict' object cannot be converted to 'Sequence'` because `ChartSpec::data` in vegafusion-core is defined as a `Vec<_>` and it looks like all the tests use the `"data" : [...]` form for the spec.

In practice it looks like vegafusion doesn't support vega-lite [data](https://vega.github.io/vega-lite/docs/data.html) attribute which is a dict?

**Comment by jonmmease** [comment]
Is this blocking for you right now @boydgreenfield?

**Comment by boydgreenfield** [comment]
> Is this blocking for you right now @boydgreenfield?

We have an open issue tracking it @jonmmease, but I've only seen it "in the wild" once so far. Note @Keats and I are working on this together and he may be retracting/editing the above comments in the morning (some test file mixup, so I do think this is a valid bug report).

**Comment by jonmmease** [comment]
Ok, thanks for the context. I'll get to it over the next couple of weeks. I think the check itself can just be removed, but I need to dig into why I added it in the first place.

**Comment by Keats** [comment]
Yeah that's what I did as well on my branch https://github.com/Keats/vegafusion but I was digging in the various libraries to see why it was there

**Comment by jonmmease** [comment]
Could you open a PR with that change @Keats? I'll dig into whether there are any downstream implications of it when I review it. Thanks!

BTW, I'm a big fan of Zola and use it for my personal website!

**Comment by jonmmease** [comment]
Done in https://github.com/vega/vegafusion/pull/494
